### PR TITLE
fix: division-by-zero in gradient calculation

### PR DESCRIPTION
**Pull request by brawncode**
### Description

updated the code to ensure we never divide by zero. using `clamp(min=1e-8)` keeps the math correct and won’t affect cases where `mask.sum() > 0`. this change prevents potential NaNs in gradients.

### Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: example_trainer/grpo.py [hunk 1]
```diff
@@ -371,7 +371,8 @@ def train(config: TrainingConfig):
             with torch.no_grad():
                 pos = (advantages > 0).float()
                 neg = (advantages <= 0).float()
-                avg_logp = (logp_per_token * mask).sum(-1) / mask.sum(-1)
+                mask_sum = mask.sum(-1).clamp(min=1e-8)
+                avg_logp = (logp_per_token * mask).sum(-1)
```
Comment (dmahan93) [inline example_trainer/grpo.py:375]: you did not divide by mask_sum here

**Review by dmahan93 — CHANGES_REQUESTED** [review]
